src/consumer/base_consumer.rs

```diff
@@ -81,6 +81,26 @@ impl<C: ConsumerContext> BaseConsumer<C> {
         Ok(())
     }
 
+    /// Pause consuming of this consumer.
+    pub fn pause(&self) {
+        unsafe {
+            let mut tp_list = rdkafka::rd_kafka_topic_partition_list_new(0);
+            rdkafka::rd_kafka_assignment(self.client.native_ptr(), &mut tp_list);
+            rdkafka::rd_kafka_pause_partitions(self.client.native_ptr(), tp_list);
+            rdkafka::rd_kafka_topic_partition_list_destroy(tp_list);
+        }
+    }
+
+    /// Resume consuming of this consumer.
+    pub fn resume(&self) {
+        unsafe {
+            let mut tp_list = rdkafka::rd_kafka_topic_partition_list_new(0);
+            rdkafka::rd_kafka_assignment(self.client.native_ptr(), &mut tp_list);
```

Comment:
I'm looking into a possible bug here at the moment: It seems like the consumer doesn't continue from the correct offsets.